src/test/kotlin/racingcar/domain/AnalyzerTest.kt

```diff
@@ -0,0 +1,173 @@
+package racingcar.domain
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+
+import org.junit.jupiter.api.Test
+import racingcar.data.Car
+
+class AnalyzerTest {
+
+    private lateinit var analyzer: Analyzer
+
+    @BeforeEach
+    fun setUp() {
+        analyzer = Analyzer()
+    }
+
+    @Test
+    fun `자동차가 움직일 수 있는지 확인하는 함수에 0을 입력하면 false를 반환한다`() {
+        // given
+        val pickedNum = 0
+
+        // when
+        val actual = analyzer.isMoveAllowed(pickedNum)
+
+        // then
+        val expected = false
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `자동차가 움직일 수 있는지 확인하는 함수에 3을 입력하면 false를 반환한다`() {
+        // given
+        val pickedNum = 3
+
+        // when
+        val actual = analyzer.isMoveAllowed(pickedNum)
+
+        // then
+        val expected = false
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `자동차가 움직일 수 있는지 확인하는 함수에 4을 입력하면 true를 반환한다`() {
+        // given
+        val pickedNum = 4
+
+        // when
+        val actual = analyzer.isMoveAllowed(pickedNum)
+
+        // then
+        val expected = true
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `자동차가 움직일 수 있는지 확인하는 함수에 9을 입력하면 true를 반환한다`() {
+        // given
+        val pickedNum = 9
+
+        // when
+        val actual = analyzer.isMoveAllowed(pickedNum)
+
+        // then
+        val expected = true
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `우승자 판별 함수에 3칸 전진한 A, 2칸 전진한 B, 0칸 전진한 C를 입력하면 A를 반환한다`() {
+        // given
+        val a = Car("A")
+        repeat(3) {
+            a.move()
+        }
+        val b = Car("B")
+        repeat(2) {
+            b.move()
+        }
+        val c = Car("C")
```

Comment:
위에서 move 함수에 distance 라는 디폴트 인자를 넣어 테스트 시에 용이하도록 수정한다면 아래와 같이 repeat이 반복되는 코드를 줄일 수 있을 것 같아요.
```suggestion
        // given
        val a = Car("A").also { it.move(3) } 
        val b = Car("B").also { it.move(2) }
        val c = Car("C")
```